### PR TITLE
UI detail

### PR DESCRIPTION
**Pull request by JustAnOtherCoder13**
new branch on top on navigation, delete navigation and replace by this one.

### CODE REVIEW

**Comment by JustAnOtherCoder13** [comment]
everything implemented work well.
good for me
missing : filter menu, connect button with view model action, add photo screen and i think it's good for phone ui.

**Comment by JustAnOtherCoder13** [comment]
everything good for me

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java [hunk 1]
```diff
@@ -1,24 +1,198 @@
...
+        super.onViewCreated(view, savedInstanceState);
+        mIsPropertyValuesAreNull = Objects.requireNonNull(mPropertyViewModel.getSelectedProperty.getValue()).getAddress() == null;
+
+        initView();
+        initDropDownMenu();
+        initClickListener();
+        configureOnClickRecyclerView();
+
+        mPropertyViewModel.isUpdateComplete.observe(getViewLifecycleOwner(), isUpdateComplete -> {
+            if (!isUpdateComplete) return;
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java:70]: `selected property` is a empty Property object ?

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java [hunk 2]
```diff
@@ -1,24 +1,198 @@
...
+                });
+    }
+
+    //___________________________________HELPERS_____________________________________________
+
+    private void setTextForCustomView(@NonNull AddPropertyInformationCustomView customView, String text) {
+        EditText editText = customView.findViewById(R.id.add_property_information_custom_view_value);
+        editText.setText(text);
+    }
+
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java]: This methpd should be defined inside the custom view itself.

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java [hunk 3]
```diff
@@ -1,24 +1,198 @@
...
+    private void initAddMediaClickListener() {
+        RecyclerViewItemClickListener.addTo(mBinding.addPropertyMediaLayout.detailCustomViewRecyclerView, R.layout.fragment_add_property)
+                .setOnItemClickListener((recyclerView, position, v) -> {
+                    if (position == 0)
+                        Toast.makeText(requireContext(), "add", Toast.LENGTH_SHORT).show();
+                });
+    }
+
+    //___________________________________HELPERS_____________________________________________
+
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java:160]: This method is part of the CustomView not the fragment.

**Review by fodeguirassy — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/PropertyDetailFragment.java [hunk 4]
```diff
@@ -4,22 +4,69 @@
...
+        setTextForCustomView(detailInformationLayout.fragmentDetailLocationCustomView, property.getAddress());
+        setTextForCustomView(detailInformationLayout.fragmentDetailNumbersOfRoomsCustomView, String.valueOf(property.getNumberOfRooms()));
+        setTextForCustomView(detailInformationLayout.fragmentDetailNumbersOfBedroomsCustomView, String.valueOf(property.getNumberOfBedrooms()));
+        setTextForCustomView(detailInformationLayout.fragmentDetailNumbersOfBathroomsCustomView, String.valueOf(property.getNumberOfBathrooms()));
+
+        descriptionTextView.setText(property.getDescription());
+        addressTextView.setLines(6);
+        addressTextView.setSingleLine(false);
+    }
+
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/PropertyDetailFragment.java:63]: It's a CustomView instance method

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/main/MainActivity.java [hunk 5]
```diff
@@ -25,15 +27,67 @@
...
+                R.drawable.ic_custom_view_update_24
+                : R.drawable.ic_custom_view_save_24);
+
+        if (isForUpdate) mUpdateButton.setOnClickListener
+                (v -> mNavController.navigate
+                        (R.id.action_propertyDetailFragment_to_addPropertyFragment));
+    }
+
+    private void initTopAppBar() {
+        ImageButton addPropertyBtn = mBinding.topAppBar.findViewById(R.id.top_bar_add_property);
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/main/MainActivity.java]: Navigation logic should be defined inside the TopAppBar CustomView. It exposes an instance method such as `onBackButtonClicked` which takes an optional lambda as a parameter. The default behavior of this method is `findNavController().popBackStack`. When needed you can override that behavior with your own back navigation logic by passing a lambda ta AppToolBar

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java [hunk 6]
```diff
@@ -13,20 +14,24 @@
...
-        super(context, attrs, defStyleAttr, defStyleRes);
-    }
+        ImageView addPropertyInformationIcon = findViewById(R.id.add_property_information_custom_view_icon);
+        TextView addPropertyInformationTitle = findViewById(R.id.add_property_information_custom_view_title);
+        //EditText addPropertyInformationEditText = findViewById(R.id.add_property_information_custom_view_value);
 
+        TypedArray attributes = getContext().obtainStyledAttributes(attrs,R.styleable.AddPropertyInformationCustomView);
+
+        addPropertyInformationIcon.setImageDrawable(attributes.getDrawable(R.styleable.AddPropertyInformationCustomView_icon));
+        addPropertyInformationTitle.setText(attributes.getText(R.styleable.AddPropertyInformationCustomView_title));
```
Comment (fodeguirassy) [inline app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java]: Remove dead code

File: app/src/main/res/layout/fragment_add_property_sold_layout.xml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="@id/add_property_sold_checkbox"
+        app:layout_constraintBottom_toBottomOf="@id/add_property_sold_checkbox"/>
+
+    <com.google.android.material.checkbox.MaterialCheckBox
+        android:id="@+id/add_property_sold_checkbox"
+        style="@style/Widget.FragmentAddPropertySoldCheckBox"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintStart_toEndOf="@id/add_property_sold_title"
```
Comment (fodeguirassy) [inline app/src/main/res/layout/fragment_add_property_sold_layout.xml:21]: format

File: app/src/main/res/layout/fragment_add_property_sold_layout.xml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        android:visibility="gone"
+        android:id="@+id/add_property_sold_edit_text"
+        style="@style/Widget.FragmentAddPropertySoldEditText"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:autofillHints="@string/dd_mm_yyyy"
+        android:hint="@string/dd_mm_yyyy"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@id/add_property_sold_checkbox"
+        app:layout_constraintTop_toTopOf="parent"
```
Comment (fodeguirassy) [inline app/src/main/res/layout/fragment_add_property_sold_layout.xml]: format

File: app/src/main/res/layout/fragment_add_property_sold_layout.xml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        />
+
+    <FrameLayout
+        style="@style/basicSeparatorStyle"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/separator_height"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/add_property_sold_title"
```
Comment (fodeguirassy) [inline app/src/main/res/layout/fragment_add_property_sold_layout.xml]: format

File: app/src/main/res/values/attrs.xml [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    </declare-styleable>
+
+    <declare-styleable name="DetailInformationCustomView">
+        <attr name="icon"/>
+        <attr name="title" />
+    </declare-styleable>
+
+    <declare-styleable name="AddPropertyInformationCustomView">
+        <attr name="icon"/>
+        <attr name="title"/>
```
Comment (fodeguirassy) [inline app/src/main/res/values/attrs.xml:16]: If it doesn't work then you can turning it into a public field you can get and set outside of the CustomView.

File: app/src/main/res/values/styles.xml [hunk 11]
```diff
@@ -2,75 +2,197 @@
...
-    <style name="Widget.TextViewTitleStyle" parent="Widget.MaterialComponents.TextView">
+    <style name="Widget.Headline1TextView" parent="Widget.MaterialComponents.TextView">
         <item name="android:paddingBottom">@dimen/basic_marge</item>
-        <item name="android:textAppearance">@style/TextAppearance.RealEstateManagerStyles.Headline4</item>
+        <item name="android:layout_marginStart">@dimen/basic_marge</item>
+        <item name="android:textAppearance">
+            @style/TextAppearance.RealEstateManagerStyles.Headline1
+        </item>
+    </style>
+
```
Comment (fodeguirassy) [inline app/src/main/res/values/styles.xml]: The parent here should be `MaterialComponents.Subtitle1` not TextView.
Also you cannot name textAppearences as `Widget.Subtitle1TextView`. TextAppearence is not a widget.

File: app/src/main/res/values/styles.xml [hunk 12]
```diff
@@ -2,75 +2,197 @@
...
     <!--__________________________________CUSTOM VIEW__________________________________-->
 
-    <style name="customViewInformationIconStyle" >
+    <style name="customViewIconStyle">
         <item name="android:paddingEnd">@dimen/basic_marge</item>
-        <item name="android:paddingStart">0dp</item>
-        <item name="srcCompat">@drawable/ic_fragment_detail_area_24</item>
     </style>
 
-    <!--TITLE-->
```
Comment (fodeguirassy) [inline app/src/main/res/values/styles.xml:34]: Same here. TextAppearence is not a widget.

File: app/src/main/res/values/styles.xml [hunk 13]
```diff
@@ -2,75 +2,197 @@
...
         <item name="android:paddingEnd">0dp</item>
-        <item name="android:paddingStart">15dp</item>
         <item name="android:textColor">@color/white</item>
         <item name="android:background">@color/black</item>
-        <item name="android:alpha">0.5</item>
+        <item name="android:alpha">0.6</item>
+    </style>
+
+    <!--__________________________________TOP BAR__________________________________-->
+
```
Comment (fodeguirassy) [inline app/src/main/res/values/styles.xml:87]: Use MaterialComponentsButton instead of AppCompat.

**Review by fodeguirassy — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java (hunk 1 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java:70]: Not exactly but is empty if come from add button and filled if come from update button

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java (hunk 3 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java:160]: Do i have to create setter in AddPropertyCustomView? If yes, how to bind the specific instance of the custom view to set the good edit text, (AreaEditText, PriceEditText...)

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/PropertyDetailFragment.java (hunk 4 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/PropertyDetailFragment.java:63]: ok done

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java (hunk 3 above)
Comment (fodeguirassy) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java:160]: Tag the CustomView. With for example an enum of `InputType`. Use it to identify the right form field and set logic.

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/main/MainActivity.java (hunk 5 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/main/MainActivity.java]: done, don't know if it's what you expect but work well

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java (hunk 6 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/utils/customView/AddPropertyInformationCustomView.java]: wich line? seem to need everything for me.

File: app/src/main/res/layout/fragment_add_property_sold_layout.xml (hunk 8 above)
Comment (JustAnOtherCoder13) [reply app/src/main/res/layout/fragment_add_property_sold_layout.xml]: if i format autofill hint and hint, create a warning on edit text, do i have to format anyway and ignore warning in layout?

File: app/src/main/res/layout/fragment_add_property_sold_layout.xml (hunk 7 above)
Comment (JustAnOtherCoder13) [reply app/src/main/res/layout/fragment_add_property_sold_layout.xml:21]: done

File: app/src/main/res/values/attrs.xml (hunk 10 above)
Comment (JustAnOtherCoder13) [reply app/src/main/res/values/attrs.xml:16]: work fine, i have forgot to bind attribute in the custom view code.

File: app/src/main/res/values/styles.xml (hunk 11 above)
Comment (JustAnOtherCoder13) [reply app/src/main/res/values/styles.xml]: ok done for all style

File: app/src/main/res/values/styles.xml (hunk 12 above)
Comment (JustAnOtherCoder13) [reply app/src/main/res/values/styles.xml:34]: done

File: app/src/main/res/values/styles.xml (hunk 13 above)
Comment (JustAnOtherCoder13) [reply app/src/main/res/values/styles.xml:87]: done

File: app/src/main/res/layout/fragment_add_property_sold_layout.xml (hunk 9 above)
Comment (JustAnOtherCoder13) [reply app/src/main/res/layout/fragment_add_property_sold_layout.xml]: done

File: app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java (hunk 3 above)
Comment (JustAnOtherCoder13) [reply app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java:160]: ok done with static method, don't think that's what you expected but i have don't find ressources to explain on internet, so do it like i think. work fine

**Pull request merged**